sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java

```diff
@@ -214,18 +215,57 @@
     @SuppressWarnings("StaticInitializerReferencesSubClass")
     public static final Configuration NONE = new NoopConfiguration();
 
+    /*
+     * Gets the global configuration shared by all client libraries.
+     */
+    private static final Configuration GLOBAL_CONFIGURATION = new Configuration();
+
     private final ConcurrentMap<String, String> configurations;
+    private final ConfigurationSource[] configurationSources;
+    private final String sectionName;
+    private final Configuration base;
 
     /**
      * Constructs a configuration containing the known Azure properties constants.
      */
     public Configuration() {
         this.configurations = new ConcurrentHashMap<>();
+        this.configurationSources = CONFIG_SOURCES;
+        this.sectionName = null;
+        this.base = null;
         loadBaseConfiguration(this);
     }
 
-    private Configuration(ConcurrentMap<String, String> configurations) {
+    public Configuration(ConfigurationSource... additionalSources) {
+        this.configurations = new ConcurrentHashMap<>();
+
+        // TODO(configuration): priority, should we add env var sources by default
+        this.configurationSources = new ConfigurationSource[CONFIG_SOURCES.length + additionalSources.length];
+
+        // reverse order, so last config is requested first to ensure priority
```

Comment:
Hmm, I'm not a fan of this as it requires explicit understanding of an API to do correctly. I'd lean towards maintaining the order the configuration stores were passed as the order they are checked.